cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2115,7 +2118,9 @@ macro(build_substrait)
   set(SUBSTRAIT_INCLUDES ${SUBSTRAIT_CPP_DIR} ${PROTOBUF_INCLUDE_DIR})
 
   add_library(substrait STATIC ${SUBSTRAIT_SOURCES})
-  set_target_properties(substrait PROPERTIES POSITION_INDEPENDENT_CODE ON)
+  set_target_properties(substrait
+                        PROPERTIES POSITION_INDEPENDENT_CODE ON
+                                   COMPILE_OPTIONS "${SUBSTRAIT_SUPPRESSED_FLAGS}")
```

Comment:
Could you use `target_compile_options()` instead?

We may be able to remove `COMPILE_OPTIONS` source property:
https://github.com/apache/arrow/blob/fe32a7dfe5e22e7737198476fe1ac0e8a5dccef2/cpp/cmake_modules/ThirdpartyToolchain.cmake#L2077-L2078